compiler/src/iree/compiler/Dialect/LinalgExt/Utils/IndexingUtils.cpp

```diff
@@ -37,12 +37,10 @@ void AttentionOpDetail::inferFromIndexingMaps(AffineMap qMap, AffineMap kMap,
   llvm::SmallDenseSet<int64_t> vSet = findPermutationsIndexingOperand(vMap);
   llvm::SmallDenseSet<int64_t> oSet = findPermutationsIndexingOperand(oMap);
 
-  // B = (Q & K & O) U (K & V & O)
+  // B = (Q & V) U (K & O)
```

Comment:
Interesting. Is this true. cc @Groverkss

Reply:
Yes. We can actually make it just (Q & V) or just (K & O) also because they are the same.

K := batch x k1 x k2
O := batch x m x n

so their intersection has to be the batch dimensions.

Similarily,

Q := batch x m x k1
V := batch x k2 x n

so their intersection has to be the batch dimension

Interestingly, this actually made me realize we are doing GQA wrong. GQA is a broadcast on the M dimension, which is why it's only on Q.